cmd/otel-allocator/server/server.go

```diff
@@ -155,32 +216,48 @@ func RemoveRegexFromRelabelAction(jsonConfig []byte) ([]byte, error) {
 // configurations such that the underlying prometheus marshaling is used. After that, the YAML is converted
 // in to a JSON format for consumers to use.
 func (s *Server) UpdateScrapeConfigResponse(configs map[string]*promconfig.ScrapeConfig) error {
-	var configBytes []byte
-	configBytes, err := yaml.Marshal(configs)
-	if err != nil {
-		return err
-	}
-	var jsonConfig []byte
-	jsonConfig, err = yaml2.YAMLToJSON(configBytes)
-	if err != nil {
-		return err
+	marshalSecretValues := []bool{false}
+	if s.httpsServer != nil {
+		marshalSecretValues = append(marshalSecretValues, true)
 	}
 
-	jsonConfigNew, err := RemoveRegexFromRelabelAction(jsonConfig)
-	if err != nil {
-		return err
+	for _, marshalSecretValue := range marshalSecretValues {
+		config.MarshalSecretValue = marshalSecretValue
+		configBytes, err := yaml.Marshal(configs)
+		if err != nil {
+			return err
+		}
+
+		var jsonConfig []byte
+		jsonConfig, err = yaml2.YAMLToJSON(configBytes)
+		if err != nil {
+			return err
+		}
+
+		jsonConfigNew, err := RemoveRegexFromRelabelAction(jsonConfig)
+		if err != nil {
+			return err
+		}
+
+		s.mtx.Lock()
+		if marshalSecretValue {
+			s.ScrapeConfigMarshalledSecretResponse = jsonConfigNew
+		} else {
+			s.scrapeConfigResponse = jsonConfigNew
+		}
+		s.mtx.Unlock()
```

Comment:
We can marshal both by setting `marshalSecretValues := []bool{false, true} ` and removing
```
if s.httpsServer != nil {
    marshalSecretValues = append(marshalSecretValues, true)
}
```
The loop is there to avoid duplication.

Reply:
If we want to avoid duplication, maybe it's better to factor it out into a function? I don't find the loop approach very readable, personally.